Spigot-API-Patches/0231-Damage-with-any-cause.patch

```diff
@@ -0,0 +1,66 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: NeumimTo <chleba48@gmail.com>
+Date: Wed, 28 Oct 2020 18:42:07 +0100
+Subject: [PATCH] Damage with any cause
+
+
+diff --git a/src/main/java/org/bukkit/entity/Damageable.java b/src/main/java/org/bukkit/entity/Damageable.java
+index fc4d3bcd9b16097086fef7975274d825b65adb10..ab0b3c1c0be56fa550c4c1787db293e149144d31 100644
+--- a/src/main/java/org/bukkit/entity/Damageable.java
++++ b/src/main/java/org/bukkit/entity/Damageable.java
+@@ -1,6 +1,8 @@
+ package org.bukkit.entity;
+ 
+ import org.bukkit.attribute.Attribute;
++import org.bukkit.event.entity.EntityDamageEvent;
++import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
+ /**
+@@ -23,6 +25,46 @@ public interface Damageable extends Entity {
+      */
+     void damage(double amount, @Nullable Entity source);
+ 
++    // Paper start
++
++    /**
++     * Deals the given amount of damage to this entity with specific DamageCause.
++     *
++     * Following damage types require to specify an entity, which triggered the damage, for such cases use
++     * @see Damageable#damage(double, EntityDamageEvent.DamageCause, Entity), otherwise they will fallback to Generic damage type
++     *
++     * DamageCause.ENTITY_ATTACK
++     * DamageCause.ENTITY_SWEEP_ATTACK
++     * DamageCause.PROJECTILE
++     * DamageCause.THORNS
++     *
++     * @param amount Amount of damage to deal
++     * @param damageCause Type of damage
++     */
++    void damage(double amount, @NotNull EntityDamageEvent.DamageCause damageCause);
++
++    /**
++     * Deals the given amount of damage to this entity, with specific DamageCause, from a specified
++     * entity.
++     *
++     * @see Damageable#damage(double, EntityDamageEvent.DamageCause)
++     *
++     * Eventually if the source is not null, then the method will also handle cases
```

Comment:
Did you mean causes?

Reply:
No, i meant cases, Plural of "case", if that do not make any sence i'll change it

Reply:
Causes with a U; you're listing causes underneath.